src/index.js

```diff
@@ -54,6 +54,15 @@ function map(array, callback) {
   return result;
 };
 
+function reduce(array, callback, initial) {
+  let accumulator = initial ?? array[0];
+  const startIndex = initial ? 0 : 1;
```

Comment:
This line is suspect 🔍 
```suggestion
  const startIndex = initial ? 0 : 1;
```